File: lua/connect_april.lua
local connect_april = {}

function connect_april.recv_request()
    local n = arg[#arg]
    local filePath = '.' .. n .. '.input'

    local file = io.open(filePath, 'r')
    if file then
        local data = file:read('*a')
        file:close()
        return json.decode(data)
    else
        print('Error reading file:', filePath)
        return nil
    end
end

function connect_april.send_response(response)
    local n = arg[#arg]
    local filePath = '.' .. n .. '.output'

    local file, err = io.open(filePath, 'w')
    if file then
        file:write(json.encode(response))
        file:close()
    else
        print('Error writing file:', err)
    end
end

return connect_april

File: ruby/connect_april.rb
# connect_april.rb

require 'json'

module ConnectApril
  def self.recv_request
    n = ARGV[-1]
    file_path = ".#{n}.input"

    begin
      data = File.read(file_path)
      JSON.parse(data)
    rescue StandardError => e
      puts "Error reading file: #{file_path}"
      nil
    end
  end

  def self.send_response(response)
    n = ARGV[-1]
    file_path = ".#{n}.output"

    begin
      File.write(file_path, JSON.generate(response))
    rescue StandardError => e
      puts "Error writing file: #{file_path}"
    end
  end
end

